subsys/net/lib/lwm2m/lwm2m_obj_portfolio.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_obj_portfolio
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <stdint.h>
+#include <init.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_resource_ids.h"
+
+#define PORTFOLIO_VERSION_MAJOR 1
+#define PORTFOLIO_VERSION_MINOR 0
+
+#define PORTFOLIO_IDENTITY_ID			0
+#define PORTFOLIO_GET_AUTH_DATA_ID		1
+#define PORTFOLIO_AUTH_DATA_ID			2
+#define PORTFOLIO_AUTH_STATUS_ID		3
+
+#define PORTFOLIO_MAX_ID 4
+
+/* Identity max based on LwM2M v1.1 conformance test requiremnts */
+#define PORTFOLIO_IDENTITY_MAX 4
+#define PORTFOLIO_AUTH_DATA_MAX 4
+/* Support only 1 even this multi instance object */
+#define MAX_INSTANCE_COUNT	1
+
+/*
+ * Calculate resource instances as follows:
+ * start with PORTFOLIO_MAX_ID
+ * subtract MULTI resources and execute because their counts include 0 resource (4)
+ * add PORTFOLIO_IDENTITY_ID resource instances
+ * add PORTFOLIO_AUTH_DATA_ID resource instances
+ */
+#define RESOURCE_INSTANCE_COUNT        (PORTFOLIO_MAX_ID - 3 + \
+					PORTFOLIO_IDENTITY_MAX + \
+					PORTFOLIO_AUTH_DATA_MAX)
+
+
+static struct lwm2m_engine_obj portfolio;
+static struct lwm2m_engine_obj_field fields[] = {
+	OBJ_FIELD_DATA(PORTFOLIO_IDENTITY_ID, RW, STRING), /* Mandaory Multi Instance */
+	OBJ_FIELD_EXECUTE(PORTFOLIO_GET_AUTH_DATA_ID), /* Optional */
+	OBJ_FIELD_DATA(PORTFOLIO_AUTH_DATA_ID, R_OPT, OPAQUE), /* Optional Multi instance */
+	OBJ_FIELD_DATA(PORTFOLIO_AUTH_STATUS_ID, R_OPT, U8), /* Optional: Single */
+};
+
+static struct lwm2m_engine_obj_inst inst[MAX_INSTANCE_COUNT];
+static struct lwm2m_engine_res res[MAX_INSTANCE_COUNT][BUTTON_MAX_ID];
```

Comment:
_PORTFOLIO_IDENTITY_MAX_, not _BUTTON_MAX_ID_

Reply:
yep